aws-datastore/src/test/java/com/amplifyframework/datastore/syncengine/OrchestratorTest.java

```diff
@@ -146,17 +149,32 @@ public void itemsPlacedInStorageArePublishedToNetwork() throws AmplifyException
 
     /**
      * Verify preventing concurrent state transitions from happening.
+     * @throws AmplifyException Not expected.
+     * @throws InterruptedException Not expected.
      */
     @Test
-    public void preventConcurrentStateTransitions() {
+    public void preventConcurrentStateTransitions() throws AmplifyException, InterruptedException {
         // Arrange: orchestrator is running
         orchestrator.start();
 
         // Try to start it in a new thread.
-        new Thread(() -> orchestrator.start()).start();
+        CountDownLatch startOrchestratorInNewThreadLatch = new CountDownLatch(1);
+        new Thread(() -> {
+            try {
+                orchestrator.start();
+                startOrchestratorInNewThreadLatch.countDown();
+            } catch (DataStoreException exception) {
+                // No need to do anything here.  Test will fail down below when latch times out.
+            }
+        }).start();
+
         // Try to start it again on a current thread.
         orchestrator.start();
 
+        if (!startOrchestratorInNewThreadLatch.await(2, TimeUnit.SECONDS)) {
+            fail("Failed to start orchestrator on a background thread.");
+        }
```

Comment:
How about:
```suggestion
        Completable
            .create(subscriber -> {
                new Thread(() -> {
                   try {
                       orchestrator.start();
                       subscriber.onComplete();
                   } catch (DataStoreException failure) {
                       subscriber.onFailure(failure);
                   }
                }).start();
            })
            .blockingAwait(5, TimeUnit.SECONDS);
```
Or:
```suggestion
        Await
            .<VoidResult, DataStoreException>result((onResult, onError) -> {
                new Thread(() -> {
                   try {
                       orchestrator.start();
                       onResult.accept(VoidResult.instance());
                   } catch (DataStoreException failure) {
                       onError.accept(failure);
                   }
                }).start();
            });
```

Reply:
I like both of these better, thanks for the suggestions.  I'll go with the RxJava approach :).   RxJava feels cleaner than using a `CountDownLatch`.    I'd like to start using the [RxJava timeout operator](http://reactivex.io/documentation/operators/timeout.html) more too, instead of CountDownLatch, where possible.